lms/djangoapps/teams/tests/test_views.py

```diff
@@ -0,0 +1,454 @@
+        self.assertEqual(4, teams['count'])
+        self.assertIn('coal team', [team['name'] for team in teams['results']])
+
+    # Text search is not yet implemented, so this should return HTTP
+    # 400 for now
+    def test_list_teams_filter_text_search(self):
+        self.get_teams_list(400, data={'text_search': 'foobar'})
+
+    @ddt.data(
+        ({}, ['Nuclear Team', 'solar team', 'Wind Team']),
+        ({'order_by': 'name'}, ['Nuclear Team', 'solar team', 'Wind Team']),
+        ({'order_by': 'open_slots'}, ['Wind Team', 'Nuclear Team', 'solar team']),
+    )
+    @ddt.unpack
+    def test_list_teams_order_by(self, data, names):
+        teams = self.get_teams_list_json(data=data)
+        self.assertEqual(3, teams['count'])
+        self.assertEqual([team['name'] for team in teams['results']], names)
+
+    # Last activity is not yet implemented, so this should return HTTP 400 for now
+    def test_list_teams_order_by_last_activity(self):
+        self.get_teams_list(400, data={'order_by': 'last_activity'})
+
+    @ddt.data({'course_id': 'foobar/foobar/foobar'}, {'topic_id': 'foobar'})
+    def test_list_team_no_results(self, data):
+        self.get_teams_list(404, data=data)
+
+    def test_list_team_page_size(self):
+        result = self.get_teams_list_json(data={'page_size': 2})
+        self.assertEquals(2, result['num_pages'])
+
+    def test_list_team_page(self):
+        result = self.get_teams_list_json(data={'page_size': 1, 'page': 3})
+        self.assertEquals(3, result['num_pages'])
+        self.assertIsNone(result['next'])
+        self.assertIsNotNone(result['previous'])
+
+    def build_team_data(self, name="Test team", course=None, description="Filler description", **kwargs):
+        """
+        Creates the payload for creating a team. kwargs can be used to specify additional fields.
+        """
+        data = kwargs
+        course = course if course else self.test_course_1
+        data.update({
+            'name': name,
+            'course_id': str(course.id),
+            'description': description,
+        })
+        return data
+
+    def post_create_team(self, expected_status, data, user=None):
+        """
+        Posts data to the team creation endpoint as user. Verifies expected_status.
+        """
+        user = user if user else self.student_user_enrolled
+        self.client.login(username=user.username, password=self.test_password)
+        response = self.client.post(reverse('teams_list'), data=data)
+        self.assertEquals(response.status_code, expected_status)
+        return response
+
+    def post_create_team_json(self, expected_status, data, user=None):
+        """
+        Posts data to the team creation endpoint as user. Verifies expected_status and returns the result as a dict.
+        """
+        response = self.post_create_team(expected_status, data, user=user)
+        return json.loads(response.content)
+
+    def test_create_team_anonymous(self):
```

Comment:
Nit: I find it a little hard to read all of these very similar tests that are variants of each other. I recommend using DDT like Christina did with the User API view tests like this one:

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/accounts/tests/test_views.py#L250